### ISSUE TITLE
boost 1.69

### ISSUE THREAD

**Issue by mr-j0nes** [open-license-manager/open-license-manager#60]
Hi there,

just wanted to report an error I had during compilation. I run Ubuntu 18.04 with boost 1.69 and I get following errors:

/usr/bin/c++   -pipe -fmessage-length=0 -std=c++11 -Wuninitialized -fPIC -O3 -DNDEBUG  -pthread -static-libstdc++ -Wl,--strip-all -static-libstdc++ -rdynamic CMakeFiles/os_linux_test.dir/Os_Linux_test.cpp.o  -o os_linux_test ../../lib/libos.a -Wl,-Bstatic -lboost_date_time -lboost_unit_test_framework -lboost_program_options -lboost_system -lboost_filesystem ../../lib/libbase.a -lcrypto -Wl,-Bdynamic -lz -ldl 
CMakeFiles/os_linux_test.dir/Os_Linux_test.cpp.o: In function `_GLOBAL__sub_I_Os_Linux_test.cpp':
Os_Linux_test.cpp:(.text.startup+0x26): undefined reference to `boost::unit_test::unit_test_log_t::instance()'
Os_Linux_test.cpp:(.text.startup+0x58): undefined reference to `boost::unit_test::decorator::collector_t::instance()'
Os_Linux_test.cpp:(.text.startup+0xea): undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(boost::unit_test::test_case*, boost::unit_test::decorator::collector_t&, unsigned long)'
Os_Linux_test.cpp:(.text.startup+0x117): undefined reference to `boost::unit_test::decorator::collector_t::instance()'
Os_Linux_test.cpp:(.text.startup+0x192): undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(boost::unit_test::test_case*, boost::unit_test::decorator::collector_t&, unsigned long)'
Os_Linux_test.cpp:(.text.startup+0x1bf): undefined reference to `boost::unit_test::decorator::collector_t::instance()'
Os_Linux_test.cpp:(.text.startup+0x23a): undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(boost::unit_test::test_case*, boost::unit_test::decorator::collector_t&, unsigned long)'
Os_Linux_test.cpp:(.text.startup+0x267): undefined reference to `boost::unit_test::decorator::collector_t::instance()'
Os_Linux_test.cpp:(.text.startup+0x2e2): undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(boost::unit_test::test_case*, boost::unit_test::decorator::collector_t&, unsigned long)'

If I compile against boost 1.65 using the -DBOOST_ROOT= flag, then it compiles fine.

Regards..

**Comment by gcontini** [comment]
Can you please explain what you're doing exactly?
Ubuntu 18.04 comes with Boost 1.65 if you install it from packages it should work fine, even without setting BOOST_ROOT (see installation guide and .travis.yml file).

If you compile boost by yourself, of course you need to set BOOST_ROOT and i guess you'd better debug cmake to be sure it finds all the required components. Also you need to be sure you compile boost with the right options... eg: compile static libraries, the right architecture...

Boost is only used by the license generator, not by the library (see example.exe) so it makes very little sense to update it, except if you want to contribute...

**Comment by mr-j0nes** [comment]
Hi,
thanks for the quick reply. Sorry for not clarifying.

As you say the default boost version for Ubuntu 18.04 is 1.65. But I had to upgrade to 1.67 manually, that is what I have in my system now. All tools that I compile and use boost use this version.  When I try to compile your tool with this version of boost I get the reported errors .

So then I downloaded the tarball of boost version 1.65 to test your tool, I compiled this boost version and when I compile your tool using BOOST_ROOT it compiles successfully.

I just want to say that your tool seem not to be fully compatible with boost 1.69.

Regards,
